qemu/tests/nic_bonding.py

```diff
@@ -77,34 +87,42 @@ def run(test, params, env):
                     time.sleep(random.randint(1, 30))
                     session_serial.cmd(link_set_cmd % (ifname, "up"))
                     time.sleep(random.randint(1, 30))
-
-        except Exception:
+        except aexpect.ShellProcessTerminatedError:
             transfer_thread.join(suppress_exception=True)
             raise
         else:
             transfer_thread.join()
 
+        logging.info('Cleaning temp file on host')
+        os.remove(host_path)
         logging.info("Failover test 2 with file transfer")
         transfer_thread = utils_misc.InterruptedThread(
-            utils_test.run_file_transfer, (test, params, env))
+            vm.copy_files_from, (guest_path, host_path))
         transfer_thread.start()
         try:
             nic_num = len(ifnames)
-            up_index = 0
+            index = 0
             while transfer_thread.isAlive():
-                up_index = up_index % nic_num
-                for num in range(nic_num):
-                    if num == up_index:
-                        session_serial.cmd(link_set_cmd % (ifnames[num], "up"))
-                    else:
-                        session_serial.cmd(link_set_cmd % (ifnames[num], "down"))
-                time.sleep(random.randint(1, 5))
-                up_index += 1
-        except Exception:
+                index = index % nic_num
+                for i in range(nic_num):
+                    session_serial.cmd(link_set_cmd % (ifnames[i], "up"))
```

Comment:
Each interface will be activated at first, and then 3 of them will be disconnected.
Any reason to change this logic? `up_index` will be `0-3`, only one of the four is always activated.

Reply:
HI @PaulYuuu, In the original logic, four interfaces will be closed at the same time, which does not meet our requirements. So I changed it to the current usage and closed any three ports to ensure that one port can be opened.

Reply:
With your modification, four interfaces will be activated simultaneously in the same subnet, this will also affect network communication.

Since the nic_num is fixed, I recommend that after getting up_index, set it up and then close others. In this way, only 2 interfaces are active at most.
```python
while transfer_thread.isAlive():
    nic_indexes = list(range(nic_num))
    up_index = up_index % nic_num
    session_serial.cmd(link_set_cmd % (ifnames[up_index], "up"))
    nic_indexes.remove(up_index)
    for num in nic_indexes:
        session_serial.cmd(link_set_cmd % (ifnames[num], "down"))
```

Reply:
ok, thanks for your help, I have changed it and test pass.